lib/classes/ConfigSchemaHandler/normalizeAjvErrors.test.js

```diff
@@ -0,0 +1,218 @@
+                              required: ['path', 'method'],
+                              additionalProperties: false,
+                            },
+                          ],
+                        },
+                      },
+                      required: ['http'],
+                      additionalProperties: false,
+                    },
+                  ],
+                },
+              },
+            },
+          },
+          'additionalProperties': false,
+        },
+        additionalProperties: false,
+      },
+    },
+    additionalProperties: false,
+  };
+
+  const resolveNormalizedErrors = config => {
+    const validate = resolveValidate(schema);
+    validate(config);
+    if (!validate.errors) return [];
+    return normalizeAjvErrors(validate.errors, config, schema);
+  };
+
+  let errors;
+  before(() => {
+    errors = resolveNormalizedErrors({
+      foo: 'bar',
+      package: { incclude: ['./folder'] },
+      functions: {
+        'invalid name': {},
+        'foo': {
+          events: [
+            {
+              bar: {},
+            },
+            {
+              http: null,
+              method: 'GET',
+            },
+            {
+              http: {
+                other: 'foo',
+              },
+            },
+            {
+              http: 'gets foo',
+            },
+            {
+              http: {
+                method: 'get',
+              },
+            },
+          ],
+        },
+      },
+    });
+  });
+
+  describe('Reporting', () => {
+    it('should report error for unrecognized root property', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== '') return false;
+          if (error.keyword !== 'additionalProperties') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for unrecognized deep level property', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== '.package') return false;
+          if (error.keyword !== 'additionalProperties') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for invalid function name', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== '.functions') return false;
+          if (error.keyword !== 'additionalProperties') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for unrecognized event', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== ".functions['foo'].events[0]") return false;
+          if (error.keyword !== 'anyOf') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for improper indentation when defining event', () =>
```

Comment:
We do not validate indentation in any way (e.g. in JSON all config can be single line), so I think description as provided is confusing

Originally this test case was described as: _Should report error for unrecognized property at event type configuration level_, and I think it was very accurate

Why it was changed?

Reply:
Could you please provide the how this case looks in YAML config?

Please note that I also changed `functions.func.events[1]` config from 
```
            {
              http: {},
              method: 'GET',
            },
```
 to
```
            {
              http: null,
              method: 'GET',
            },
```
which is has YAML equivalent to look like this, which I call 'improper indentation' case:

```
foo:
 - http:
   method: GET
``` 

I could not imagine how YAML would look like in the initial config. Or could you please provide how YAML should look?

Reply:
> Could you please provide how this case looks in YAML config?

I think it could be:

```yaml
http:
  method: GET
path: /
```

> which has YAML equivalent to look like this, which I call 'improper indentation' case:

Thanks for clarification. It definitely makes big sense to test this case. So let's keep it (and indeed naming you propose sounds good, can you just add a comment that shows example of YAML that produces that - that will prevent confusion I was caught by)

And I would keep the other test as well, so let's test both

Reply:
Thanks for clarification. I'll add this case

Reply:
When I tried to add that spec, I realized that implementation does not fully meet it. 

When the config is like this:
```
    events:
      - http:
          method: get
          path: foo
        cors: true

```

then it produces an error `Unsupported function event 'http' at functions.someFunc.events[0]` with `anyOf` keyword and `functions.someFunc.events[0]` dataPath.

So meeting the spec would require changing in implementation. And I already feel quite exhausted, as this PR lasts for many months, features keep adding or changing... So if you really think the error is improper, feel free to jump in not only into spec, but also into the implementation.

Reply:
Ok, no problem @preshetin. Great thanks for outstanding work so far! I totally agree it was a lot (even though we tried to minimize this initial mvp like effort)

I'll do my best to finalize this next week

Reply:
Thank you @medikoo for your guidance. I leaned a lot when making this PR. 

I hope users would finally enjoy the benefits of validation